src/python/pants/option/global_options.py

```diff
@@ -442,7 +442,24 @@ def register_bootstrap_options(cls, register):
             ),
         )
 
-        # TODO(#7203): make a regexp option type!
+        register(
+            "--ignore-warnings",
```

Comment:
Justifications should be used sparingly and only when they really count; otherwise they weaken the argument or induce wolf cying effects or both. Presumably we do not go changing option names just to make them shorter; so the solid justification is making the option conform to most commonly useful usage. With that, the other path is to keep the existing option name and add a $literal$ marker instead. The idea that $literal$ would then be the most common usage would be what tips to making the unfortunate choice of thrashing an existing option name.

Reply:
> Presumably we do not go changing option names just to make them shorter

We do for common options that we realized were overly verbose, e.g. `--print-exception-stacktrace` became `--print-stacktrace` for no reason other than cutting down on typing.

But ack that the more important motivation is literal string matching.